lib/vsc/mympirun/pmi/sched.py

```diff
@@ -0,0 +1,177 @@
+#
+# Copyright 2019-2019 Ghent University
+#
+# This file is part of vsc-mympirun,
+# originally created by the HPC team of Ghent University (http://ugent.be/hpc/en),
+# with support of Ghent University (http://ugent.be/hpc),
+# the Flemish Supercomputer Centre (VSC) (https://www.vscentrum.be),
+# the Flemish Research Foundation (FWO) (http://www.fwo.be/en)
+# and the Department of Economy, Science and Innovation (EWI) (http://www.ewi-vlaanderen.be/en).
+#
+# https://github.com/hpcugent/vsc-mympirun
+#
+# vsc-mympirun is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation v2.
+#
+# vsc-mympirun is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with vsc-mympirun.  If not, see <http://www.gnu.org/licenses/>.
+#
+"""
+Base PMI Sched class, all actual classes should inherit from this one
+
+The role of the Sched class is mainly to construct the correct sched-specific PMI call
+"""
+
+import os
+from vsc.utils.fancylogger import getLogger
+from vsc.mympirun.common import SchedKlass
+from vsc.utils.run import run_file, async_to_stdout
+
+# simple intermediate representation of the job resources
+JOB_INFO = {
+    'tnodes': None,  # total number of nodes
+    'ncores': None,  # number of cores per node
+    'nranks': None, # number of ranks per node
+    'nmem': None,  # memory per node
+    'ngpus': None,  # number of gpus per node: None means no GPUs present, 0 means don't use GPUs
+}
+
+class Sched(SchedKlass):
+    LAUNCHER = None
+
+    def __init__(self, options=None, **kwargs):
+        if not hasattr(self, 'log'):
+            self.log = getLogger(self.__class__.__name__)
+        if not hasattr(self, 'options'):
+            self.options = options
+
+        self.sched_id = None
```

Comment:
This is not needed, `None` seems to be the default in `set_sched_id()`.

Reply:
On second thoughts, why is the `set_sched_id` function needed, it is only used here? Just removed that function and do the `os.environ.get` here?

Reply:
code removed